cardinal/ecs/tick_test.go

```diff
@@ -35,6 +40,59 @@ func TestTickHappyPath(t *testing.T) {
 	assert.NilError(t, twoWorld.LoadGameState())
 	assert.Equal(t, uint64(10), twoWorld.CurrentTick())
 }
+func TestIfPanicMessageLogged(t *testing.T) {
+
+	w := inmem.NewECSWorldForTest(t)
+	//replaces internal Logger with one that logs to the buf variable above.
+	var buf bytes.Buffer
+	bufLogger := zerolog.New(&buf)
+	cardinalLogger := ecs.Logger{
+		&bufLogger,
+	}
+	w.InjectLogger(&cardinalLogger)
+	// In this test, our "buggy" system fails once Power reaches 3
+	errorTxt := "BIG ERROR OH NO"
+	w.AddSystem(func(world *ecs.World, queue *ecs.TransactionQueue, _ *ecs.Logger) error {
+		panic(errorTxt)
+	})
+	assert.NilError(t, w.LoadGameState())
+	ctx := context.Background()
+
+	defer func() {
+		if panicValue := recover(); panicValue != nil {
```

Comment:
assert that panicValue is equal to errorTxt, so that if a panic shows up from any other source the test will fail.